banderwagon/element.go

```diff
@@ -72,36 +78,54 @@ func (p Element) BytesUncompressed() [UncompressedSize]byte {
 func BatchNormalize(elements []*Element) {
 	// The elements slice might contain duplicate pointers,
 	// dedupe them to avoid double work.
-	dedupedElements := make([]*Element, 0, len(elements))
+	mapDedupedElements := make(map[*Element]struct{}, len(elements))
 	for _, e := range elements {
-		found := false
-		for i := range dedupedElements {
-			if dedupedElements[i] == e {
-				found = true
-				break
-			}
-		}
-		if found {
-			continue
-		}
```

Comment:
When generating proofs with _a lot_ of openings, we have a very long list of `elements`. 
In L82+L78 we have a nested for loop which is `O(n^2)` and it's quite bad when we have too many elements.

So, I've changed the `dedupedElements` slice to a map, so at least is `O(n)`. 

This isn't some "theoretical optimization", I was surprised to find we spent a reasonable time on this `O(n^2)` part, so I had to fix it.